### PR TITLE
Update vcert

### PR DESCRIPTION
**Pull request by Gl4di4torRr**

### CODE REVIEW

File: pkg/certs/venafi.go [hunk 1]
```diff
@@ -35,70 +34,79 @@ func (p *VenafiProvider) Provision(host string, validFrom string, validFor time.
...
-			cert,
-			key,
-			notAfter}, nil
+	//
+	// 1.1. compose request object
+	//
+	enrollReq := &certificate.Request{
+		Subject: pkix.Name{
+			CommonName:         host,
+			Organization:       []string{"Venafi.com"},
```
Comment (Gl4di4torRr) [inline pkg/certs/venafi.go]: We need to make these generic so that consumers can input the correct values.

File: pkg/certs/venafi.go [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
-	"github.com/hashicorp/terraform/helper/resource"
-	"github.com/sirupsen/logrus"
 )
 
 type VenafiProvider struct {
 }
 
 func (p *VenafiProvider) Provision(host string, validFrom string, validFor time.Duration, isCA bool, rsaBits int, ecdsaCurve string) (keypair KeyPair, certError error) {
+	
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```
Comment (etsauer) [inline pkg/certs/venafi.go]: I don't think you want `InsecureSkipVerify: true` here. The CA should take care of the handshake.

File: pkg/certs/venafi.go [hunk 3]
```diff
@@ -35,70 +34,79 @@ func (p *VenafiProvider) Provision(host string, validFrom string, validFor time.
...
-	url := os.Getenv("VENAFI_API_URL")
-	venafi := vcert.NewClientTPP(username, password, url)
-	enrollreq := &govcert.EnrollReq{
-		CommonName: host,
-		Zone:       os.Getenv("VENAFI_CERT_ZONE")}
-
-	resp, err := venafi.Do(enrollreq)
+	//
+	// 0. get client instance based on connection config
+	//
```
Comment (Digi59404) [inline pkg/certs/venafi.go]: Can we remove these generic comments and add some of our own? These come directly from the example code Venafi provided.

Comment (Gl4di4torRr) [reply pkg/certs/venafi.go]: @Digi59404 i removed the step comments but gave the url to the example that we followed.

Comment (Digi59404) [reply pkg/certs/venafi.go]: Works for me thanks!

File: pkg/certs/venafi.go (hunk 2 above)
Comment (Gl4di4torRr) [reply pkg/certs/venafi.go]: The `InsecureSkipVerify: true` is used for the venafi certificate in this case. The consumer would have to add the venafi cert to the cert-operator trust store.

File: pkg/certs/venafi.go [hunk 4]
```diff
@@ -35,70 +44,98 @@ func (p *VenafiProvider) Provision(host string, validFrom string, validFor time.
...
-		cert = []byte(respmap["Certificate"].(string))
-
-		if err != nil {
-			return KeyPair{}, NewCertError("Failed to create certificate: " + err.Error())
-		}
-		return KeyPair{
-			cert,
-			key,
-			notAfter}, nil
+	var tlsTppConfig = &vcert.Config{
```
Comment (Gl4di4torRr) [inline pkg/certs/venafi.go]: @etsauer this is the config to that sets the trust bundle.

File: pkg/certs/venafi.go [hunk 5]
```diff
@@ -35,70 +44,98 @@ func (p *VenafiProvider) Provision(host string, validFrom string, validFor time.
...
+
+var pp = func(a interface{}) {
+	b, err := json.MarshalIndent(a, "", "    ")
+	if err != nil {
+		fmt.Println("error:", err)
+	}
+	log.Println(string(b))
+} 
+
+func getSSLFlag() string {
```
Comment (Gl4di4torRr) [inline pkg/certs/venafi.go]: I have not incorporated this yet... but I was thinking we could just set ssl on or off via the env var.

**Comment by etsauer** [comment]
@Gl4di4torRr can you include in the comments what things need to be tested and how?

**Comment by Gl4di4torRr** [comment]
@etsauer @Digi59404 okay, here is how to test

----
export OPERATOR_NAME=cert-operator
export VENAFI_API_URL=https://myvenafi.com/api
export VENAFI_USER_NAME=myusername
export VENAFI_PASSWORD=mypassword
export VENAFI_CERT_ZONE='myzone'
export CERT_OP_CONFIG=/etc/cert-operator/config.yaml
export VENAFI_CERT_PATH=/path/to/cert
oc login ...
oc new-project cert-operator
oc process -f deploy/rbac.yaml -p NAMESPACE=cert-operator | oc apply -f-
operator-sdk up local
----

**Pull request merged**